src/cli/telemetry/telemetry.ts

```diff
@@ -0,0 +1,293 @@
+      await action();
+    } catch (e) {
+      error = e;
+    }
+
+    const end = new Date();
+    duration = end.getTime() - start.getTime();
+  }
+
+  // We'll get componentCount details inside the taskBuild, so let's not send two messages.
+  if (!tracking || (flags.task == 'build' && !flags.args.includes('--watch'))) {
+    return;
+  }
+
+  const session_id = await getTelemetryToken();
+  const packages = await getInstalledPackages();
+  const targets = await getActiveTargets(config);
+
+  const data: TrackableData = {
+    session_id,
+    yarn: isUsingYarn(),
+    duration,
+    targets,
+    packages: packages || [],
+    arguments: flags.args,
+    task: flags.task,
+    stencil: coreCompiler.version,
+    system: `${sys.name} ${sys.version}`,
+    os_name: details.platform,
+    os_version: details.release,
+    cpu_model: `${details.cpuModel}`,
+    build: coreCompiler.buildId,
+    typescript: versions.typescript,
+    rollup: versions.rollup,
+  };
+
+  await sendMetric('stencil_cli_command', data);
+  logger.debug(`${logger.blue('Telemetry')}: ${logger.gray(JSON.stringify(data))}`);
+
+  if (error) {
+    throw error;
+  }
+}
+
+function isUsingYarn() {
+  return process.env.npm_execpath.includes('yarn');
+}
+
+async function getActiveTargets(config: Config): Promise<string[][] | OutputTargetOptions[][]> {
+  return (config.outputTargets as any[]).map(target => {
+    let outputTargetSettings: OutputTargetOptions = {};
+
+    if (target.type === 'www') {
+      outputTargetSettings.serviceWorker = isObject(target?.serviceWorker);
+    }
+
+    return [target.type, outputTargetSettings];
+  });
+}
+
+async function getInstalledPackages() {
+  try {
+    // Read package.json and package-lock.json
+    const appRootDir = process.cwd();
+
+    const packageJson: PackageJsonData = await tryFn(
+      readJson,
+      getCompilerSystem().resolvePath(appRootDir + '/package.json'),
+    );
+    const packageLockJson: any = await tryFn(
+      readJson,
+      getCompilerSystem().resolvePath(appRootDir + '/package-lock.json'),
+    );
+
+    const packages = Object.entries({
+      ...packageJson.devDependencies,
+      ...packageJson.dependencies,
+    });
+
+    // Collect packages only in the stencil, ionic, or capacitor org's:
+    // https://www.npmjs.com/org/stencil
+    const ionicPackages = packages.filter(
+      ([k]) => k.startsWith('@stencil/') || k.startsWith('@ionic/') || k.startsWith('@capacitor/'),
+    );
+
+    // Original method, but cross reference with package-lock.json, don't consider yarn.
+    const versions = packageLockJson
+      ? ionicPackages.map(
+          ([k, v]) =>
+            `${k}@${packageLockJson.dependencies[k].version || packageLockJson.devDependencies[k].version || v}`,
+        )
+      : ionicPackages.map(([k, v]) => `${k}@${v}`);
+
+    return versions;
+  } catch (err) {
+    hasDebug() && console.error(err);
+    return [];
+  }
+}
+
+export async function shouldTrack(ci: boolean) {
```

Comment:
This function looks like it's only used in this file, so we can probably remove `export` here
```suggestion
async function shouldTrack(ci: boolean) {
```